### ISSUE TITLE
MOPAC2SDF  multiple MOPAC optimized structures  to SDF

### ISSUE THREAD

**Issue by tobigithub** [openbabel/openbabel#1789]
- [ ] I believe this to be a bug with Open Babel
- [x] This is a feature request

## Environment Information
Open Babel version: 2.4.1
Operating system and version: all

---
There was a recent interesting paper about the "Sobering Assessment of Small-Molecule Force Field
Methods for Low Energy Conformer Predictions"(https://arxiv.org/ftp/arxiv/papers/1705/1705.04308.pdf) 

So I was trying to utilize that information from the paper above and instead of using MMFF94 I optimized my MOPAC file containing 1000 structures with openmopac PM7 and basically had a mopac output file  (*.out *.arc) with a 1000 optimized PM7 structures.

Now in order to utilized them further in my pipeline I needed to convert them from MOPAC output format (*.out *.arc) back to SDF. With the current GUI it only converts one single file, but not multiple MOPAC optimized structures. Would it be possible to integrate such a feature or is that rather something one would do outside with a small tools (similar to SDF2SMI) or maybe there is already a tool or some python code?

Tobias

**Comment by ghutchis** [comment]
Yes, that's my paper (http://doi.org/10.1002/qua.25512) - I wasn't aware that MOPAC had multi-molecule output files.

It wouldn't take much to patch, but as @baoilleach said, it would be great to have an example file (e.g. water and CO2 or something like that).

It wouldn't be hard to write a Python script, but the patch would be much easier.

As far as the protocol from the paper, please let me know if this is something that interests you academically, and I'd be happy to give tips.

**Comment by tobigithub** [comment]
@baoilleach - please see example below. The only trick to let MOPAC calculate multiple molecules in a single file, is to separate multiple molecules by an empty line CR/LF. 

@ghutchis - thank you, its great to read something like this, especially for people bit outside the field like me. MMFF94 is just so embedded in the minds, and DFT on 1000 molecules is still to far away, but PM7 runs literally in seconds, so really doable (besides all the conversion issues).  For docking one could also use PM6-D3H4 which is a bit more accurate and maybe COSMO if needed.

The general workflow I use is
1) SMILES to 3D SDF (add H and generate MMFF94 low energy conformer)
2) 3D SDF to XYZ (XYZ needed for input in other tools or DFT calculations)
3) XYZ to MOPAC cartesian (need to add MOPAC keywords and single CR/LF separator)
4) MOPAC ARC/OUT to optimized SDF (needed for pipeline and additional handling)

Now one can process 100k molecules in a single file and then parse the output like
IP, HOMO, LUMO, COSMO volume, pkA and other values. Best would be of course
to have all the calculated properties added as SDF fields but that is another story.
(One could use the https://cactus.nci.nih.gov/SDF_toolkit/ to add or remove properties)

**Example MOPAC input file with three molecules:** 
````
PM7 OPT XYZ
Methane

C   1.05860 1 -0.08080 1 -0.00520 1
H   2.15080 1 -0.08080 1 -0.00520 1
H   0.69460 1 -1.10850 1 -0.07110 1
H   0.69460 1  0.49010 1 -0.86220 1
H   0.69460 1  0.37590 1  0.91770 1

PM7 OPT XYZ
Formaldehyde

C   1.10270 1 -0.08080 1 -0.00520 1
O   2.32730 1 -0.08080 1 -0.00520 1
H   0.51500 1 -1.01080 1 -0.06480 1
H   0.51500 1  0.84910 1  0.05450 1

PM7 OPT XYZ
Water

O   1.04450 1 -0.06090 1 -0.00390 1
H   2.01240 1 -0.10720 1 -0.00690 1
H   0.76570 1 -0.98700 1 -0.06330 1
````

The MOPAC output is a *.arc (small) and *.out (large) MOPAC output file.
When using the OpenBabel GUI only the last molecule in the *.out is converted
Attached are the example files from SMI to SDF to XYZ to MOP to ARC to OUT.

[Mopac-SDF-Converter.zip](https://github.com/openbabel/openbabel/files/1790228/Mopac-SDF-Converter.zip)
